rwgame/GameConfig.cpp

```diff
@@ -155,6 +155,7 @@ bool GameConfig::parseConfig(
                     } else {
                       success = false;
                       RW_MESSAGE(e.what());
+                      return;
                     }
                 }
```

Comment:
I think this entire code path could be nicer with an early out:
```
try {
   sourceValue = srcTree.get<config_t>(key, translator);
} catch (pt::ptree_bad_path &e) {
   if (!optional) {
      success = false;
      RW_MESSAGE(e.what());
      return;
   }
   sourceValue = defaultValue;
}
```
(Makes it more obvious that we abort when the argument is mandatory but failed to be parsed)